homeassistant/config.py

```diff
@@ -828,18 +845,35 @@ async def merge_packages_config(
 
 
 async def async_process_component_config(  # noqa: C901
-    hass: HomeAssistant, config: ConfigType, integration: Integration
+    hass: HomeAssistant,
+    config: ConfigType,
+    integration: Integration,
+    *,
+    raise_on_failure: bool = False,
```

Comment:
This parameter is causing a lot of duplicated code, it's also not useful when the caller in real wants to reload a single platform because it will cause us to abort on the first platform which has an error.

Instead, maybe we should:
- Always raise if parsing the component configuration fails
- Collect errors caught when setting up the platforms and return them

Reply:
> Always raise if parsing the component configuration fails
I think that is better indeed.

Reply:
Hmm, `async_integration_yaml_config` is called with the parameter to allow raising an exception to avoid the need to re-raise in the integration. Do you want to remove the parameter from `async_integration_yaml_config`?

Reply:
This is no longer relevant, I'm marking this as resolved 👍